### ISSUE TITLE
target renders null after unmount/mount

### ISSUE THREAD

**Issue by shrugs** [smooth-code/react-teleporter#9]
## 💬 Questions and Help

Hi! I'm looking for some help with some odd behavior I've encountered.

Here's the behavior I'm seeing. The setup, without drowning anyone in code, is:

There is a `Root` component. There is a `Parent` component (which is rendered by `Root`) that contains a `Teleporter.Target` and is in charge of rendering `A` and `B` (using the simple router mentioned below). The `OldParent` is like `Parent` in that it renders a `.Target`

1) unmount `OldParent` and mount `Parent` which renders a `Teleporter.Target` and mounts `A`, which renders a `Teleporter.Source` (ok)
2) within `Parent`, unmount `A` and mount `B` which contains no `Teleporter.Source` (ok)
3) unmount `B` and (re) mount `A` 👉 results in null teleportation
...
4) if I unmount the `Parent` component going back to `OldParent` and then re-mount it (which re-mounts/renders `A`) the teleportation works as expected.

I wrote my own baby router component which is what's rendering these A and B components based on some in-memory state. I don't think the router is relevant to the bug, but I mention it because it certainly might be. The rendering boils down to

```jsx
const Route = useMemo(() => views[last(history).key], [views, history]);

...
// <Parent /> looks like
return (
  <View>
    <Route />
  </View>
)

// where <View /> basically looks like
<>
  {children}
  <Teleporter.Target as="footer" />
</>
```

---

The furthest I got while debugging is that in subsequent renders of `Teleporter.Source`, `context.element` is null so the `Source` renders null due to https://github.com/smooth-code/react-teleporter/blob/master/src/index.js#L36

throwing a debugger into the `useCallback()` block, though, shows that `context.element` is set to null during the unmount of a `PreviousComponent` (expected) but then immediately set to the correct element when `Parent` is mounted, never being called again. When the source in `A` attempts to render the second time, though, the `element` state is null because `context.element` is null within the `useLayoutEffect` hook.

**Comment by gregberge** [comment]
Hello @shrugs, your problem looks complex. The code of `react-teleporter` is very simple. I suggest you to add it directly in your project and to try to debug it. If the bug is confirmed, feel free to submit a PR.

**Comment by gregberge** [comment]
Fixed in 9a5611e9ba0cf66a4d646360413dad5b4d90fdab
